### DISCUSSION TITLE
Disposition Assessment Review

### DISCUSSION THREAD

**Discussion by jwildfire** [Gilead-Public/gsm#39] [General]
@gwu05 @jwildfire to review gsmu::Disp_Assess() and consider what if any changes need to be made in migration to gsm

**Comment by jwildfire** [comment]
Met with @gwu05 @mattroumaya  and @dsanders2gilead  to spec out revised disposition assessment. Updated Assessment workflow below. * indicates that a new function is needed: 

-*Disp_Map creates dfInput for a specific discontinuation type/reason (e.g. Study Discontinuation because of Withdrawn consent)
- Transform_EventCount() creates a site level discontinuation and participation counts .
- *Analyze_FisherExact() runs analysis when method="Fisher"
- *Analyze_ChiSq()* runs analysis when method="ChiSq"
- Flag() flags sites
- Summarize() creates data summary

Disp_Map should support different types of discontinuation and any reason specified. (Note: Don't think we'll really need different map functions for AdAM and Raw here). Consider the following parameters:
- dfDisp - disposition data set 
- strCol - Column with discontinuation reason
- strReason - Reason for discontinuation found in strCol (Should support "Any" which flags any reason. 

Output should be one record per person with `Count` column equal to 0/1 indicating whether the participant was discontinued for the reason specified. 

Analyze functions can be modeled after Wilcoxon (#29) once it is ready.

**Reply by mattroumaya** [reply]
Took a first pass at this, and apologize in advance if I'm missing anything important. 

I'd appreciate any info about the raw data that would typically be used for this because I don't think I covered this, and am unsure where to pull from in `clindata`

- The input data below was taken from [gsmu testing for Disposition_Map](https://github.com/Gilead-BioStats/gsmu/blob/dev/tests/testthat/test_Disposition_Map.R#L20-L25)
- `strReason` accepts a character vector of reasons, in case someone would want to run the assessment on a number of different discontinuation reasons (but not all). I'm not sure if this should strictly accept a single reason or "any" - can easily be changed if that's the case.
- `strReason` is also effectively case insensitive, checks equality using `tolower()`
- Code below should be able to run locally to test out and provide any feedback. 

```
library(tidyverse)
library(safetyData)

# using safetyData example from gsmu

dfADSL2 <- safetyData::adam_adsl %>% 
  mutate(SUBJID = paste0("00000-", SUBJID),
         TR01SDT = TRTSDT, TR01EDT = TRTEDT, 
         COMT01FL = COMP8FL, DCT01RS = ifelse(COMT01FL=="N",DCREASCD,""), 
         EOSDT = RFENDT, 
         COMSFL = ifelse(DISCONFL=="Y","N",""), 
         DCSREAS = ifelse(DCREASCD=="Completed","",DCREASCD))


# Disp_Map function
Disp_Map <- function( dfDisp = NULL, strCol, strReason = "any") {
  # DCSREAS == "Completed"?
  dfInput <- dfDisp %>%
    filter(SAFFL == "Y" | !is.na(!!strCol)) %>%
    select(SubjectID = SUBJID,
           SiteID = SITEID,!!strCol) %>%
    mutate(SubjectID =  str_remove(SubjectID, ".*-"))
  
  
  if ("any" %in% tolower(strReason)) {
    dfInput <- dfInput %>%
      mutate(Count = 1)
    
  } else {
    dfInput <- dfInput %>%
      mutate(Count = ifelse(tolower(!!sym(strCol)) %in% tolower(strReason), 1, 0))
    
  }
  
  return(dfInput)
  
}

# output
disp <- Disp_Map(
  dfDisp = dfADSL2, 
  strCol = "DCREASCD", 
  strReason = c("adVerSe eVeNt", "sponsor DECISION")
  )


  SubjectID SiteID DCREASCD         Count
  <chr>     <chr>  <chr>            <dbl>
1 1015      701    Completed            0
2 1023      701    Adverse Event        1
3 1028      701    Completed            0
4 1033      701    Sponsor Decision     1
5 1034      701    Completed            0
6 1047      701    Adverse Event        1
```

**Comment by jwildfire** [comment]
Start with `Analyze_Wilcoxon()` as a placeholder analysis function. We'll write up the other methods soon.

**Comment by jwildfire** [comment]
Next step is to create `Disp_Assess()` using `AE_Assess` as a template

**Comment by mattroumaya** [comment]
Started looking at this and am not sure `Analyze_Wilcoxon()` will work as a placeholder (example below). Happy to take a closer look and see if I can write a fisher exact/chi square function, but would need to get a better understanding before going in that direction.

cc: @jwildfire @gwu05 

``` r
library(gsm)

dfDisp <- gsm::Disp_Map(dfDisp = safetyData::adam_adsl, strCol = "DCREASCD", strReason = "adverse event")

lAssess <- list()
lAssess$dfDisp <- dfDisp
lAssess$dfTransformed <- gsm::Transform_EventCount( lAssess$dfDisp, cCountCol = "Count")

names(lAssess$dfTransformed)
#> [1] "SiteID"     "N"          "TotalCount"

# doesn't work because Analyze_Wilcoxon() requires columns: "SiteID", "N", "TotalExposure", "TotalCount", "Rate"
lAssess$dfAnalyzed <- gsm::Analyze_Wilcoxon( lAssess$dfTransformed)
#> Error in gsm::Analyze_Wilcoxon(lAssess$dfTransformed): all(c("SiteID", "N", "TotalExposure", "TotalCount", "Rate") %in%  .... is not TRUE
```

**Reply by gwu05** [reply]
@mattroumaya Yeah, you're right, as written Analyze_Wilcoxon will not work due to the stopifnot. We need to generalize Analyze_Wilcoxon to be able to specify any numeric column in a dfTransformed dataset for analysis for it to work directly, rather than have it focused exclusively for use in the AE/lab abnormality assessments

**Comment by mattroumaya** [comment]
Initial pass for [`Disp_Assess()` is ready](https://github.com/Gilead-BioStats/gsm/tree/fix-153-disp-assess) with some obvious limitations, but should be a good point to build off of or for someone else to grab and develop. 

cc @gwu05 @jwildfire - happy to continue with this with some more guidance/detail for the other `Analyze_X` functions
